F-FrontEnd/src/F-compile-expr.c

```diff
@@ -1097,10 +1097,15 @@ compile_ident_expression(expr x)
         goto done;
     }
 
-
     if(ID_CLASS(id) == CL_PARAM){
-        if(VAR_INIT_VALUE(id) != NULL) 
-            return VAR_INIT_VALUE(id);
+        if ((id = declare_variable(id)) != NULL) {
```

Comment:
Line 1101 to line 1103 seems to have no effects. `tp` isn't used until line 1108.
And if `declare_variable(id)` returns `NULL`, it will cause a segmentation fault.